### PR TITLE
Add option to enable health check logging

### PR DESCRIPTION
**Pull request by tnguyen14**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate . 
Error: 
Unsupported block type
  on main.tf line 82, in resource "google_compute_health_check" "tcp":
  82:   
log_config
 {
Blocks of type "log_config" are not expected here.
Error: 
Unsupported block type
  on main.tf line 106, in resource "google_compute_health_check" "http":
 106:   
log_config
 {
Blocks of type "log_config" are not expected here.
terraform_validate ./examples/minimal 
Error: 
Invalid value for module argument
  on main.tf line 77, in module "test_ilb":
  77:   health_check = 
local.health_check
The given value is not suitable for child module variable "health_check"
defined at ../../variables.tf:73,1-24: attribute "enable_log" is required.
terraform_validate ./examples/simple 
Error: 
Invalid value for module argument
  on main.tf line 40, in module "gce-ilb":
  40:   health_check = 
local.health_check
The given value is not suitable for child module variable "health_check"
defined at ../../variables.tf:73,1-24: attribute "enable_log" is required.
terraform_validate ./test/fixtures/minimal 
Error: 
Invalid value for module argument
  on ../../../examples/minimal/main.tf line 77, in module "test_ilb":
  77:   health_check = 
local.health_check
The given value is not suitable for child module variable "health_check"
defined at ../../../variables.tf:73,1-24: attribute "enable_log" is required.
terraform_validate ./test/setup 
Success!
 The configuration is valid.
 ```

**Comment by tnguyen14** [comment]
I tried using the `google-beta` provider, but the tests still failed. I am not sure why.

**Comment by morgante** [comment]
The examples would also need to be updated.

**Comment by tnguyen14** [comment]
@morgante happy to update it once we can get this to work/ tests to pass.

**Comment by morgante** [comment]
@tnguyen14 This won't pass until the examples are updated, because the tests use the examples.

**Comment by tnguyen14** [comment]
@morgante I tried to update the example, but it's still failing. Is there a way I can look at the error message to figure out what might be wrong?

**Comment by morgante** [comment]
@tnguyen14 [This comment](https://github.com/terraform-google-modules/terraform-google-lb-internal/pull/49#issuecomment-812202759) gets updated with the latest results.

**Comment by tnguyen14** [comment]
Test is passing now.

File: main.tf [hunk 1]
```diff
@@ -78,12 +79,17 @@ resource "google_compute_health_check" "tcp" {
     port_name    = var.health_check["port_name"]
     proxy_header = var.health_check["proxy_header"]
   }
+
+  log_config {
```
Comment (morgante) [inline main.tf:83]: We should make this a dynamic block.

```suggestion
  dynamic "log_config" {
    for_each = var.health_check["enable_log"] ? [true] : []
    content {
      enable = true
    }
  }
```

**Review by morgante — CHANGES_REQUESTED** [review]

**Review by morgante — APPROVED** [review]

**Pull request merged**